.github/workflows/migrate.yml

```diff
@@ -105,13 +105,36 @@ jobs:
           fi
 
           $JUJU switch source-controller
+          $JUJU version
+          
+          # Retry `juju migrate` a few times. It can take a while for the
```

Comment:
Let's not do it this way. 
migrate must work first time, every time if the model is ready
Let's poll the model for idle active and then migrate

Reply:
doesn't work, because even if the machine is marked as "running", the machine agent could still be marked as "pending" in the DB. This is when you get the error
```
ERROR source prechecks failed: machine 0 agent not functioning at this time (pending)
```
These things can sometimes take a couple of seconds to update and as you know, Juju can't guarantee the order of certain events, just eventual consistency

Reply:
ok you're right, we can do it this way